frontend/src/app/App.test.tsx

```diff
@@ -917,6 +923,25 @@ describe("App.onHistoryChange", () => {
       expect(instance.onPageChange).toHaveBeenLastCalledWith("sub_hash")
     })
   })
+
+  it("doesn't push a new history when the new url contains an anchor so we don't rerun", () => {
+    pushStateSpy = jest.spyOn(window.history, "pushState")
+    window.history.pushState({}, "", "/#foo=bar")
+
+    const instance = wrapper.instance() as App
+    instance.handleNewSession(new NewSession(NEW_SESSION_JSON))
+
+    expect(window.history.pushState).toHaveBeenLastCalledWith(
+      {},
+      "",
+      "/#foo=bar"
+    )
+
+    // check that onPageChange is not run when anchor is in the url
+    waitFor(() => expect(instance.onPageChange).not.toHaveBeenCalled())
```

Comment:
I'm fairly certain this test doesn't work the way it's intended to. As written you can't tell if the `waitFor` completes because the `onHistoryChange` function chooses not to call `instance.onPageChange` or if the test simply gets to this line before `onHistoryChange` has a chance to run.

I think the most straightforward way of fixing this would be to just call `onHistoryChange` directly (it shouldn't matter that the call to `pushState` causes it to be called another time asynchronously) and dropping the `waitFor`.